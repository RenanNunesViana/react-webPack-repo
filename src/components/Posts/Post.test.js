import { render, screen, describe, it, expect } from '@testing-library/react';
import { Posts } from '.';
import React from 'react'

const props = {
  posts: [
    { title: 'test 1', cover: 'http//:test/1', body: 'little test 1', id: 1 },

    { title: 'test 2', cover: 'http//:test/2', body: 'little test 2', id: 2 },
    { title: 'test 3', cover: 'http//:test/3', body: 'little test 3', id: 3 },
  ],
};

describe('<Posts/>', () => {
  it('should has all posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /test/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /test/i })).toHaveLength(3);
    expect(screen.getAllByText(/little/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /test 3/i })).toHaveAttribute('src', 'http//:test/3');
  });

  it('should not have any posts', () => {
    const { container } = render(<Posts />);
    expect(container.firstChild).toBeEmptyDOMElement();
  });

  it('should match snap shot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
