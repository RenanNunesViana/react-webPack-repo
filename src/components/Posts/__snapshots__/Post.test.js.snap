// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snap shot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="test 1"
      src="http//:test/1"
    />
    <div
      class="post-content"
    >
      <h2>
        test 1
         
        1
      </h2>
      <p>
        little test 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="test 2"
      src="http//:test/2"
    />
    <div
      class="post-content"
    >
      <h2>
        test 2
         
        2
      </h2>
      <p>
        little test 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="test 3"
      src="http//:test/3"
    />
    <div
      class="post-content"
    >
      <h2>
        test 3
         
        3
      </h2>
      <p>
        little test 3
      </p>
    </div>
  </div>
</div>
`;
